src/openslide-decode-jpeg.c

```diff
@@ -345,6 +346,20 @@ static bool jpeg_decode(struct _openslide_file *f,  // or:
       return false;
     }
 
+    // we only override the JPEG color space if the file has an unknown set of
+    // component_ids and we have been given a known colorspace
+    if (space != JCS_UNKNOWN && cinfo->num_components == 3) {
+      int cid0 = cinfo->comp_info[0].component_id;
+      int cid1 = cinfo->comp_info[1].component_id;
+      int cid2 = cinfo->comp_info[2].component_id;
+
+      if (cid0 == 0 && cid1 == 1 && cid2 == 2) {
+        // SVS sometimes mistakenly uses 0/1/2 to mean RGB, eg. CMU-2.svs
+        // sample file
+        cinfo->jpeg_color_space = space;
```

Comment:
As discussed above, we should override the input space whenever it isn't specified as `JCS_UNKNOWN`.

Reply:
ok, done